Library/Homebrew/os/mac/xcode.rb

```diff
@@ -139,7 +139,8 @@ def uncached_version
       end
 
       def provides_autotools?
-        version < "4.3"
+        # Xcode 2.5's autotools are too old to rely on at this point
+        (version < "4.3") && (version > "2.5")
```

Comment:
Looks like that was done in d22ad92a8479501b080312dbc1570476cda8172f, and the deps themselves were moved to compat in d108bf0a55446df39592604bbfefa641d9c0e734. I can drop this commit and make a followup PR that removes this method (and places that call it) and/or moves it to compat.